### PR TITLE
Add paddle.nn.SmoothL1Loss

### PR DESCRIPTION
**Pull request by yghstill**
### PR types
 New features

### PR changes
 APIs 

### Describe
add paddle.nn.SmoothL1Loss. This operator is calculate smooth_l1_loss. Creates a criterion that uses a squared term if the absolute element-wise error falls below 1 and an L1 term otherwise. In some cases it can prevent exploding gradients. Also known as the Huber loss.

class:
```python
class paddle.nn.SmoothL1Loss(reduction='mean', delta=1.0, name=None)
```
functioanl:
```python
def paddle.nn.functioanl.smooth_l1_loss(input, label, reduction='mean', delta=1.0, name=None)
```
docs
image
image
image

Examples:
```python
           import paddle
            import numpy as np
            paddle.disable_static()
            input_data = np.random.rand(3,3).astype("float32")
            label_data = np.random.rand(3,3).astype("float32")
            input = paddle.to_tensor(input_data)
            label = paddle.to_tensor(label_data)
            loss = paddle.nn.SmoothL1Loss()
            output = loss(input, label)
            print(output.numpy())
```

### CODE REVIEW

**Review by yaoxuefeng6 — APPROVED** [review]
LGTM

**Review by jerrywgz — APPROVED** [review]

**Review by jzhang533 — APPROVED** [review]
lgtm
great to know no c++ op is introduced and huber_loss api is reused.

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM
